### DISCUSSION TITLE
GraphQL API response for commit status is null

### DISCUSSION THREAD

**Discussion by pascaldevink** [community/community#24508] [Apps, API and Webhooks]
<p>Using the GraphQL API Explorer (but also in my own code), and with the following query:</p>
<pre><code>query($owner:String!,$name:String!) {
  repository(owner:$owner,name:$name) {
    pullRequests(last:10, states:OPEN) {
      nodes {
        commits(last:1) {
          nodes {
            commit {
              status {
                state
              }
            }
          }
        }
      }
    }
  }
}
</code></pre>
<p>The commit status is always null for private repositories, even though the GraphQL API Explorer has the right access, and some of the first pull requests do return a status.</p>
<p>Am I using the wrong query to get the commit status of the last commit for a given PR, or is something else wrong?</p>

**Comment by nickvanw** [comment]
<p>Hey <a class="mention" href="/pascaldevink">@pascaldevink</a>!</p>
<p>Your query looks right to me! I tested it in a couple of my private repositories and it returned the data that I expected. In your code, are you using a Personal Access Token (PAT)? If so, what scopes does it have? </p>

**Comment by pascaldevink** [comment]
<p>Hi <a class="mention" href="/nickvanw">@nickvanw</a>, thanks for coming back to me.</p>
<p>I’m currently using a personal access token with all the <code>repo</code> access, and <code>read:org</code>, although I want to eventually migrate to using OAuth.</p>
<p>I think I’ve tracked down the root issue: For some repositories, I was able to get all the status information on commits, but for some repositories I was only able to do that up till a certain point.</p>
<p>We are using CircleCI in our organisation and in October we’ve switched to using Pull Request Checks. You can probably already guess when the status field stopped giving usefull information by now.</p>
<p>I see the checks API is still in beta, but the announcement and documentation only cover V3 of the API. Is there an equivalent in V4? And is there any documentation on it for V4?</p>

**Comment by nickvanw** [comment, marked as answer]
<p>Hey <a class="mention" href="/pascaldevink">@pascaldevink</a>!</p>
<p>I think the GraphQL API may be able to do what you want! There is a <code>Commit.checkSuites</code> connection that should get you the names and statuses of CircleCI’s Checks, something like this:</p>
<pre><code>query ($owner: String!, $name: String!) {
  repository(owner: $owner, name: $name) {
    pullRequests(last: 10, states: OPEN) {
      nodes {
        commits(last: 1) {
          nodes {
            commit {
              checkSuites(first: 10) {
                nodes {
                  app {
                    name
                  }
                  checkRuns(first:10) {
                    nodes {
                      status
                      name
                    }
                  }
                  status
                }
              }
            }
          }
        }
      }
    }
  }
}
</code></pre>
<p>Let me know if that works!</p>

**Comment by Starefossen** [comment]
<blockquote>
<p><a class="mention" href="/nickvanw">@nickvanw</a> wrote:</p>
</blockquote>
<blockquote>
<p>Hey <a class="mention" href="/pascaldevink">@pascaldevink</a>!</p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<blockquote>
<p>Your query looks right to me! I tested it in a couple of my private repositories and it returned the data that I expected. In your code, are you using a Personal Access Token (PAT)? If so, what scopes does it have? </p>
</blockquote>
<p><span class="mention">@nickvanwI</span> am seeing a similar issue, I can see status for private and public repos with no branch protection rules, however, on repos with branch protection rules on I only get null for all status checks in the Graphql API.</p>

**Reply by Starefossen** [reply]
LOL... I came back three years and found my own answer (still valid) 😅

**Comment by sfpetersen** [comment]
<p><a class="mention" href="/starefossen">@starefossen</a> hit the same exact issue in a repo with branch protection rules. Were you able to find anything?</p>

**Comment by lee-dohm** [comment]
<p>Hello <a class="mention" href="/sfpetersen">@sfpetersen</a>, thanks for joining us!</p>
<p>If you can share the query you’re using, that will generally help people diagnose the problem you’re encountering much, much easier. So if you can share your query, that would be very helpful.</p>

**Comment by sfpetersen** [comment]
<p>The specific query I was using is below, though I’m sure I could reproduce it without the associatedPullRequests bit. Thanks!</p>
<pre><code class="lang-auto">{
  repository(owner: "my-organization", name: "my-repo") {
    object(oid: "667589986fac336663fc89dce53f9c993e9bfafe") {
      ... on Commit {
        associatedPullRequests(first: 10) {
          nodes { 
            number 
            state
            title
            mergeable
            reviewDecision
            commits(last: 1){
              nodes{
                commit{
                  commitUrl
                  oid
                  statusCheckRollup {
                    state
                  }
                  status {
                    id
                  }
                }
              }
            }
          }
        }
      }
    }
  }
}
</code></pre>
<p>Interestingly, the “statusCheckRollup” field seems to work fine - but the “status” field is always null.</p>
<p>In this specific case, what I’m really looking for is a way to get the result of the “Require branches to be up to date before merging” check. I’m not actually sure if that is broken out in the status field.</p>

**Comment by lee-dohm** [comment]
<p>I was unable to reproduce the problem you’re describing. I used the query:</p>
<pre><code class="lang-auto">{
  repository(owner: "atom", name: "atom") {
    object(oid: "4a9d56f52cab00a22744858d4077600d5365197a") {
      ... on Commit {
        associatedPullRequests(first: 10) {
          nodes { 
            number 
            state
            title
            mergeable
            reviewDecision
            commits(last: 1){
              nodes{
                commit{
                  commitUrl
                  oid
                  statusCheckRollup {
                    state
                  }
                  status {
                    id
                  }
                }
              }
            }
          }
        }
      }
    }
  }
}
</code></pre>
<p>and saw <code>SUCCESS</code> in the <code>statusCheckRollup.state</code> field.</p>
<aside class="quote no-group" data-username="sfpetersen" data-post="8" data-topic="13762">
<div class="title">
<div class="quote-controls"></div>
 sfpetersen:</div>
<blockquote>
<p>In this specific case, what I’m really looking for is a way to get the result of the “Require branches to be up to date before merging” check. I’m not actually sure if that is broken out in the status field.</p>
</blockquote>
</aside>
<p>That sounds like a separate question from this entirely. Would you mind opening a new topic for specifically this question <a class="mention" href="/sfpetersen">@sfpetersen</a>?</p>

**Comment by sfpetersen** [comment]
<p>OK yes I will open a new topic when I get a chance. To be clear, I also get <code>SUCCESS</code>  in the  <code>statusCheckRollup.state</code> field, but the <code>status</code> field is <code>null</code>. It’s my understanding that <code>status</code> should have more details regarding the individual status checks.</p>

**Comment by sfpetersen** [comment]
<p><a class="mention" href="/lee-dohm">@lee-dohm</a> I opened a separate topic for the other question: <a href="https://github.community/t/api-query-to-get-result-of-require-branches-to-be-up-to-date-before-merging-branch-protection-check-for-a-pull-request/122782" class="inline-onebox">API query to get result of "Require branches to be up to date before merging" branch protection check for a Pull Request</a>. Thanks!</p>
